payjoin/src/uri/mod.rs

```diff
@@ -151,7 +153,17 @@ impl bitcoin_uri::de::DeserializationState<'_> for DeserializationState {
         match key {
             "pj" if self.pj.is_none() => {
                 let endpoint = Cow::try_from(value).map_err(|_| InternalPjParseError::NotUtf8)?;
-                let url = Url::parse(&endpoint).map_err(|_| InternalPjParseError::BadEndpoint)?;
+                #[cfg(not(feature = "v2"))]
+                let url = Url::parse(&endpoint).map_err(|e| {
+                    InternalPjParseError::BadEndpoint(error::BadEndpointError::UrlParse(e))
+                })?;
+                #[cfg(feature = "v2")]
+                let url = url_ext::parse_with_fragment(&endpoint).map_err(|_| {
+                    InternalPjParseError::BadEndpoint(BadEndpointError::FragmentParse(
+                        ParseEndpointFragmentError::LowercaseEndpointFragment,
+                    ))
```

Comment:
The BadEndpointError is being discarded, but it conveys useful information that needs to be propagated. `parse_with_fragment` can fail either with LowercaseEndpointFragment OR with a UrlParse error here. The following would let that propagate:

```suggestion
                let url = url_ext::parse_with_fragment(&endpoint).map_err(InternalPjParseError::BadEndpoint)
```